docs/docs/pages/style-spec/expressions.md

```diff
@@ -0,0 +1,211 @@
+<!--copyeditor enable however-->
+<!--copyeditor enable clear-->
+
+<!--copyeditor ignore perform-->
+Expressions perform only one kind of implicit type conversion: a data expression used in a context where a [color](#types-color) is expected will convert a string representation of a color to a color value. In all other cases, if you want to convert between types, you must use one of the _type conversion_ expression operators: [`to-boolean`](#types-to-boolean), [`to-number`](#types-to-number), [`to-string`](#types-to-string), or [`to-color`](#types-to-color). For example, if you have a feature property that stores numeric values in string format, and you want to use those values as numbers rather than strings, you can use an expression such as `["to-number", ["get", "property-name"]]`.
+
+If an expression accepts an array argument and the user supplies an array literal, that array _must_ be wrapped in a `literal` expression (see the examples below). When GL-JS encounters an array in a style-spec property value, it will assume that the array is an expression and try to parse it; the library has no way to distinguish between an expression which failed validation and an array literal unless the developer makes this distinction explicit with the `literal` operator. The `literal` operator is not necessary if the array is returned from a sub-expression, e.g. `["in", 1, ["get", "myArrayProp"]]`.
+
+```json
+// will throw an error
+{
+    "circle-color": ["in", 1, [1, 2, 3]]
+}
+
+// will work as expected
+{
+    "circle-color": ["in", 1, ["literal", [1, 2, 3]]]
+}
+```
+
+# Expression reference
+
+<!-- This h1 is an exception as the following headings need to be an h2 for scrollspy to work on the sidebar. See site.css for override styles for these headings. -->
+
+
+## Types
+
+The expressions in this section are for testing for and converting between different data types like strings, numbers, and boolean values.
+
+Often, such tests and conversions are unnecessary, but they may be necessary in some expressions where the type of a certain sub-expression is ambiguous. They can also be useful in cases where your feature data has inconsistent types; for example, you could use `to-number` to make sure that values like `"1.5"` (instead of `1.5`) are treated as numeric values.
+
+{{<ExpressionReference group='Types' />}}
```

Comment:
Mixing markdown and react components seems like an odd choice.
I would prefer using a single technology instead of this mixture.
Preferably, that the part that are "dynamic" are generated and not partially programmed using code, syntax, and libraries that are not documented...
As an initial phase, I would recommend removing these react components and simply hardcode everything into the markdown.
After that we can decide how to proceed.